CodingPractice/PythonAssignments/cleancode/EmailExtract.py

```diff
@@ -0,0 +1,19 @@
+class MalformedEmailAddressException(Exception):
+    pass
+
+
+class EmailDetails:
+    def __init__(self, id, domain):
+        self.id = id
+        self.domain = domain
+
+    def __eq__(self, other):
+        return self.id == other.id and self.domain == other.domain
+
+
+def extract_email_address_details(email_addr):
+    try:
+        email_details = email_addr.split('@')
+        return EmailDetails(email_details[0], email_details[1])
+    except:
```

Comment:
How constructing the MalformedEmailAddressException with a helpful message so that the traceback will show which email_addr caused the problems.

Reply:
As I said earlier how about
```raise MalformedEmailAddressException( email_addr )```

We already know it is an invalid email address because the type of the exception is MalformedEmailAddress to putting 'This is an invalid email address' does not really add anything.